src/sagemaker_huggingface_inference_toolkit/optimum_utils.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2023 The HuggingFace Team, Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib.util
+import logging
+import os
+
+
+_optimum_neuron = False
+if importlib.util.find_spec("optimum") is not None:
+    if importlib.util.find_spec("optimum.neuron") is not None:
+        _optimum_neuron = True
+
+logger = logging.getLogger(__name__)
+
+
+def is_optimum_neuron_available():
+    return _optimum_neuron
+
+
+def get_input_shapes(model_dir):
+    """Method to get input shapes from model config file. If config file is not present, default values are returned."""
+    from transformers import AutoConfig
+
+    input_shapes = {}
+    input_shapes_available = False
+    # try to get input shapes from config file
+    try:
+        config = AutoConfig.from_pretrained(model_dir)
+        if hasattr(config, "neuron_batch_size") and hasattr(config, "neuron_sequence_length"):
+            input_shapes["batch_size"] = config.neuron_batch_size
+            input_shapes["sequence_length"] = config.neuron_sequence_length
+            input_shapes_available = True
+            logger.info(
+                f"Input shapes found in config file. Using input shapes from config with batch size {input_shapes['batch_size']} and sequence length {input_shapes['sequence_length']}"
+            )
+    except Exception:
+        input_shapes_available = False
+
+    # return input shapes if available
+    if input_shapes_available:
+        return input_shapes
```

Comment:
Should the environment variables override the config file values instead? That seems like the more typical case to me.

Or if the current behavior is desired, consider logging a warning message if environment variables are provided but will be ignored (especially if the values are different).

Reply:
I would say no. Since the config value are only present if the model is compiled and are the source of truth. Since the model will only work with the values which are used for compilation. 
Meaning if env var would overwrite the config it would create errors. 

Like the idea of the warning, added that!